app.js

```diff
@@ -160,3 +123,73 @@ function revertIds() {
+//
+//   Actually there is also "model" logic, which is about how we represent the actual board in Javascript,
+//   how do we "model" it.
+//   You could say that is part of "core" logic, but you can also look at it as a separate "model" logic.
+//
+//   So this is called MVC (Model-View-Controller) architecture (controller == core logic), and is a popular pattern in
+//   programming, for certain tasks where it is a good fit. You can google this, although it will probably be a bit
+//   to abstractly explained, but still give it a look.
+//
+//   So you really want to work in this way, because otherwise it will be too hard (already is).
+//
+//   In our case, this would mean something like this:
+//
+//   function displayBoard(board) { ... } // Does all the `document.` stuff to show the board in the browser.
+//     // Any board though, not just initial board! That is important.
+//     // This is our "View" logic.
+//
+//   const initialBoard = [     // List of 8 lists where each of those inner lists has 8 elements. This is our "model" logic.
+//     [{ type: "rook", color: "white"}, { type: "knight", color: "white"}, ... ],
+//     [null, null, null, ...],
+//     [null, ...],
+//     ...,
+//     [{ type: "rook", color: "black"}, { type: "knight", color: "black"}, ...]
+//   ];
+//   const initialGameState = {  // This is another part of our "model" logic.
+//     currentPlayer: "white",
+//     board: initialBoard
+//   }
+//
+//   And now comes the crux of things, our "core" logic (or Controller).
+//   It will consists of bunch of functions.
+//
+//   function movePiece([startRow, startCol], [endRow, endCol], board) { ... }
+//   // This function will, given the current board, return a new board that will have the piece moved from start to end location.
+//   // We could also have one global `currentBoard` variable that this function would just update, that is alternative approach,
+//   // but this is even nicer, where you don't have global variable but are instead passing the board around and creating modifications of it.
+//
+//   function isMoveValid([startRow, startCol], [endRow, endCol], board) { ... }
+//
+//   function checkForVictory(board) { ... }
+//
+//   function calculateCurrentScore(board) { ... }
+//
+//   And so on and on.
+//
+//   The whole point is, when doing complex stuff like chess rules and logic, you dont' have to dig through `docuemnt` and classes and DOM elements and stuff, you instead always focus on your simple list of lists and all is nice and simple.
+//   So you will want to rewrite your chess like this!
+//   I would suggest you forget drag and drop for start. Just try to get initial board to draw with this MVC approach, and then provide two input fields in your program, where user says initial field and end field, liek F1 and F2 and then they can press "move", and then try to get that move happening. Later we can upgrade that to be drag and drop instead of entering moves via text.
+
```

Comment:
Sounds like a good approach! I can maybe help you kickstart it, since I understand that it is a bit abstract at the moment -> once you get going with it, I believe you will be able to continue on your own, especially after finishing this first solution.